src/main/java/io/javalin/http/JavalinResponseWrapper.kt

```diff
@@ -0,0 +1,73 @@
+package io.javalin.http
+
+import io.javalin.core.JavalinConfig
+import io.javalin.core.util.Header
+import io.javalin.core.util.Util
+import java.io.InputStream
+import javax.servlet.ServletOutputStream
+import javax.servlet.WriteListener
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+import javax.servlet.http.HttpServletResponseWrapper
+
+class JavalinResponseWrapper(res: HttpServletResponse, rwc: ResponseWrapperContext) : HttpServletResponseWrapper(res) {
+    private val outputStreamWrapper by lazy { OutputStreamWrapper(res, rwc) }
+    override fun getOutputStream() = outputStreamWrapper
+    fun write(resultStream: InputStream?) = if (resultStream != null) outputStreamWrapper.write(resultStream) else Unit
+}
+
+class ResponseWrapperContext(request: HttpServletRequest, val config: JavalinConfig) {
+    val accepts = request.getHeader(Header.ACCEPT_ENCODING) ?: ""
+    val clientEtag = request.getHeader(Header.IF_NONE_MATCH) ?: ""
+    val type = HandlerType.fromServletRequest(request)
+    val autogenerateEtags = config.autogenerateEtags
+    val compressionStrategy = config.inner.compressionStrategy
+}
+
+class OutputStreamWrapper(val res: HttpServletResponse, val rwc: ResponseWrapperContext) : ServletOutputStream() {
+
+    companion object {
+        @JvmStatic var minSize = 1500
```

Comment:
Maybe add a comment: `// 1500 by default, since nothing is gained by compressing responses smaller than this`

Reply:
Are you sure that's actually true? Seems there's a lot of back and forth discussion on this subject, with everybody giving different numbers: https://webmasters.stackexchange.com/questions/31750/what-is-recommended-minimum-object-size-for-gzip-performance-benefits

Though perhaps there's something you know about JavaX.Server/Jetty/Javalin that I am not aware of :)

Reply:
Sorry, missed this comment. I'm not sure it's true, I just read it one time when I was implementing GZIP.